### PR TITLE
sdba - Grouped detrending

### LINKED ISSUE #442: sdba - Grouped Detrending
We need to implement grouped detrending into sdba.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #442 
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
- [ ] `bumpversion (minor / major / patch)` has been called on this branch
- [ ] Tags have been pushed (`git push --tags`)

* **What kind of change does this PR introduce?** 
All detrending is now made using the `Grouper` class of sdba. The BaseDetrend exposed methods call `group.apply()` with the `_fit` and `_get_trend ` methods of the subclasses.

Performance could be lower for simple detrendings. To code it as elegantly as I could, I didn't code shortcuts. When calling `fit`, the `_fit(da)` method returns a grouped dataset. This dataset is broadcasted back along the 'time' coordinate of the dataarray to detrend and then grouped again for the trend computation (which is a non-grouping grouped operation). For something as simple as `MeanDetrend` it's a lot of overhead...

* **Does this PR introduce a breaking change?** 
Yes, in order to achieve this as elegantly as I could, I removed the resampling possibility of  `PolyDetrend`. It could be put back, but it's slightly more complicated. Watcha think @huard?

Results of DQM might be different between this new method and the previous one.

EDIT: Comment on the performance
* **Other information**:
Might not work with the current master of xarray : calling `polyfit` on an element from `groupby` could throw some errors that should be fixed with pydata/xarray#4099.

### CODE REVIEW

File: xclim/sdba/utils.py [hunk 1]
```diff
@@ -195,7 +195,7 @@ def broadcast(
     """
     if sel is None:
         sel = {}
-
+    print(group)
```
Comment (huard) [inline xclim/sdba/utils.py:198]: debugging leftovers

Comment (aulemahal) [reply xclim/sdba/utils.py:198]: darn it

File: xclim/sdba/detrending.py [hunk 2]
```diff
@@ -90,47 +126,33 @@ class PolyDetrend(BaseDetrend):
...
     def _fit(self, da, dim="time"):
-        if self.freq is not None:
-            da = da.resample(
-                time=self.freq, label="left", loffset=loffsets[self.freq]
-            ).mean()
-        self._fitds = da.polyfit(dim=dim, deg=self.degree, full=True)
+        return da.polyfit(dim=dim, deg=self.degree)
 
-    def _get_trend(self, da):
+    def _get_trend(self, grpd, dim="time"):
```
Comment (huard) [inline xclim/sdba/detrending.py:149]: I remember hard-coding `time` in a couple of places in the code. Probably worth writing a test with a `mytime` dimension to check everything work as expected.

Comment (aulemahal) [reply xclim/sdba/detrending.py:149]: Good idée

**Pull request merged**